### DISCUSSION TITLE
Sorting of columns

### DISCUSSION THREAD

**Discussion by eborre** [mytourbook/mytourbook#1117] [General]
Hello,

Back in training mode and thus increased MTB use, upgraded to latest version, happy to play with the sort feature in the flat view yet...

What is the reasoning/technical constraint behind some columns not being sortable? Tags I can understand, it's a multiple select, harder to sort, but average speed for instance, why can we not sort on that field?

Cheers,

   Eric

**Comment by wolfgang-ch** [comment]
Only columns which have a corresponding sql field in the tour database can be sorted. Average speed and pace are computed values from distance and time which is currently not stored in the database

**Reply by eborre** [reply]
Thank you.

Is this because the sorting is done by the database engine in the query?

Versus sorting in the UI layer?

Does this mean the average speed is recalculated every time? Would it be worth storing it?

E

**Reply by wolfgang-ch** [reply]
> Is this because the sorting is done by the database engine in the query?

Yes
 
> Versus sorting in the UI layer?

No

> Does this mean the average speed is recalculated every time? 

Yes

> Would it be worth storing it?

No

I just implemented the sorting for avg speed/pace https://github.com/mytourbook/mytourbook/commit/4a143fa8a7bc8bee3b47c66b94a2af6c90928fcf with custom functions in the derby database. It will be available in the next version. This sorting is slightly slower than using just native fields but the average sorting time is still below 200 ms for about 5000 tours

Are there other fields which should be sorted?

**Comment by eborre** [comment]
Good morning,

This is wonderful news.

I'm sure I could come up with ideas for other fields but none right now.

Thank you!

E
